src/extension_instance.ml

```diff
@@ -108,6 +108,45 @@ let stop_server t =
     else Promise.return ()
 ;;
 
+let check_ocaml_lsp_available sandbox =
+  let ocaml_lsp_version sandbox =
+    Sandbox.get_command sandbox "ocamllsp" [ "--version" ]
+  in
+  let cwd =
+    match Workspace.workspaceFolders () with
+    | [ cwd ] -> Some (cwd |> WorkspaceFolder.uri |> Uri.fsPath |> Path.of_string)
+    | _ -> None
+  in
+  Cmd.output ?cwd (ocaml_lsp_version sandbox)
+  |> Promise.Result.fold
+       ~ok:(fun (_ : string) -> ())
+       ~error:(fun (_ : string) ->
+         "Sandbox initialization failed: `ocaml-lsp-server` is not installed in the \
+          current sandbox.")
+;;
+
+let suggest_to_upgrade_ocaml_lsp_server () =
+  let open Promise.Syntax in
+  let upgrade_lsp_text = "Yes" in
+  let no_upgrade = "No" in
+  let+ selection =
+    Window.showInformationMessage
+      ~message:"OCaml-LSP server is not up to date. Do you want to upgrade it?"
+      ~choices:[ upgrade_lsp_text, `Update_lsp; no_upgrade, `No_upgrade ]
+      ()
+  in
+  match selection with
+  | Some `Update_lsp ->
+    let (_ : Ojs.t option Promise.t) =
+      Vscode.Commands.executeCommand
+        ~command:Extension_consts.Commands.upgrade_ocaml_lsp_server
+        ~args:[]
+    in
+    ()
```

Comment:
I think we should wait for that promise to resolve.

Reply:
It turns out that right now the binding to execute vscode commands does not allows us to really wait for that command to complete, but that's something we should consider fixing later.